core/src/bitcoin_syncer.rs

```diff
@@ -75,6 +75,12 @@ pub(crate) async fn save_block(
         block_height
     );
 
+    // update the block_info as canonical if it already exists
+    let block_id = db.set_block_as_canonical_if_exists(Some(dbtx), block_hash).await?;
```

Comment:
why is this required btw? is this the case of a reorg happening to a previously discarded alternate chain?

Reply:
also, if this is the case, shouldn't we re-process the transactions in the db?